### ISSUE TITLE
Update of NetCDF_jll to use stdlib libcurl and MbedTLS in julia 1.6

### ISSUE THREAD

**Issue by Alexander-Barth** [JuliaPackaging/Yggdrasil#2401]
Currently `NetCDF_jll` cannot be loaded in julia 1.6 beta1 because it has been build using an older version of libcurl/MbedTLS (compared to the version of these library that are used in julia 1.6) as described in https://github.com/JuliaLang/julia/issues/39015 .

```
julia> using NetCDF_jll
ERROR: InitError: could not load library "/home/ulg/gher/abarth/.julia/artifacts/877149aedda703fde4a848bffd955df8934ca876/lib/libnetcdf.so"
libmbedtls.so.12: cannot open shared object file: No such file or directory
Stacktrace: [...]
```

I tried to make this change as described here:
https://github.com/JuliaLang/julia/issues/39015#issuecomment-758515025

but the build process of NetCDF did not find `libcurl`. I am not sure what is the correct way to tell to BinaryBuilder to use a library bundeld with julia. Besides the `lib*.so` files the build process would also need the header files (at least `curl.h`, I think).

Maybe @visr or @giordano have an idea?
Many thanks!

**Comment by visr** [comment]
Not sure if it would help, but you could try submitting a PR with your changes, to see if the same things happen on the buildbots. Then you can directly add the `julia_compat="1.6"` kwarg to build_tarballs, since it would make the new build for julia 1.6+ only (like the version of LibCURL_jll it requires).

**Comment by visr** [comment]
Fixed by #2402, #2453, #2454.
